File: Final/Code/Project_4/BaiToan5/Ex1_9/ex1_9.py
def previous_sibling(parent, v):
    p = parent[v]
    result = -1
    for u in range(1, v):  
        if parent[u] == p:
            result = u
    return result

parent = [-1, -1, 1, 1, 2, 2, 3]  

for v in range(1, 7):
    print(f"previous_sibling({v}) = {previous_sibling(parent, v)}")

File: Final/Code/Project_4/BaiToan5/Ex1_9/ex1_9.cpp
#include <iostream>
#include <vector>
using namespace std;

int previous_sibling(const vector<int>& parent, int v) {
    int p = parent[v];
    int result = -1;
    for (int u = 1; u < v; ++u) {
        if (parent[u] == p) {
            result = u;
        }
    }
    return result;
}

int main() {
    vector<int> parent = {-1, -1, 1, 1, 2, 2, 3};

    for (int v = 1; v <= 6; ++v) {
        cout << "previous_sibling(" << v << ") = " 
             << previous_sibling(parent, v) << endl;
    }

    return 0;
}